### ISSUE TITLE
[TS] FormField is missing `reverse` property in TS definitions

### ISSUE THREAD

**Issue by kachar** [grommet/grommet#5430]
`FormField` component is missing `reverse` property in the type definitions although it works okay when provided

https://github.com/grommet/grommet/blob/00e0878202df5d7e0ae93bb2f2836c576f71c1eb/src/js/components/FormField/index.d.ts#L5-L36

### Expected Behavior

No error

### Actual Behavior

```
Type '{ type: string; name: string; disabled: true; value: string | undefined; component: FC<TextInputProps>; placeholder: string; reverse: true; icon: Element; }' is not assignable to type 'IntrinsicAttributes & FormFieldExtendedProps & { children?: ReactNode; }'.
  Property 'reverse' does not exist on type 'IntrinsicAttributes & FormFieldExtendedProps & { children?: ReactNode; }'.ts(2322)
```

### URL, screen shot, or Codepen exhibiting the issue

![image](https://user-images.githubusercontent.com/893608/124924048-b4562400-e003-11eb-8caa-0673bfa18f5e.png)

https://codesandbox.io/s/grommet-v2-template-forked-6dvme

### Steps to Reproduce

```tsx
<Form>
  <FormField
    reverse
    type="text"
    value="value"
    icon={<Text>Suffix</Text>}
  />
</Form>
```

### Your Environment

- Grommet version: `2.17.3`
- Browser Name and version: any
- Operating System and version (desktop or mobile): any

**Comment by rhian-cs** [comment]
Hello, may I give this issue a try?

**Comment by britt6612** [comment]
@rhian-cs that would be awesome 👏

**Comment by rhian-cs** [comment]
Hello everyone, I know this was supposed to be a simple issue, but I'm having some trouble testing the change and the warning message.

The TypeScript build doesn't seem to complain when I use the `reverse` prop, or any additional (invalid) prop.

I've made the changes for this PR (add the `reverse` definition to the typescript types file) but I've not tested it. Should I open the PR, or test it first?

**Comment by jcfilben** [comment]
Hi @rhian-cs I put together a [codesandbox](https://codesandbox.io/s/grommet-v2-template-forked-hw0m0?file=/index.tsx) that shows the issue. You should see a red underline under the reverse prop. Using the code in it you should be able to test if you fix worked

**Comment by kachar** [comment]
I've also updated the original codesandbox from the description of the issue so it reflects the problem in `index.tsx` instead of `index.js` file

https://codesandbox.io/s/grommet-v2-template-forked-6dvme

![image](https://user-images.githubusercontent.com/893608/135399611-6e9fb6ac-3aba-45d3-9aa1-30644dc67776.png)

**Comment by rhian-cs** [comment]
Hello everyone! Thanks for updating the codesandboxes!

I've been taking a look at them over the past week and I've noticed the problem lies in the different versions. I was also able to reproduce the issues locally.

- In @kachar 's [codesandbox](https://codesandbox.io/s/grommet-v2-template-forked-6dvme) the `reverse` prop is not accepted, with the grommet version being `2.16.2`.
- In @jcfilben 's [codesandbox](https://codesandbox.io/s/grommet-v2-template-forked-hw0m0?file=/index.tsx) the `reverse` prop is accepted, with the grommet version being `2.18.0`.

This means the `reverse` prop is accepted in the latest version and it works fine, but I've noticed another issue. It seems that some of the props are being accepted when they shouldn't, and some existing props have the correct type being validated as expected:

![image](https://user-images.githubusercontent.com/72531802/136102082-ff638fea-f149-4234-a556-4a5c8bde914a.png)

I can open a new issue for this, with more details. And when that's fixed, I think we can take another look at this issue. What do you think?

If you want, I can also open a PR with the `reverse` prop being defined in `index.d.ts`, but I theoretically wouldn't be able to test if TypeScript accepts it or not.

**Comment by jcfilben** [comment]
Hi @rhian-cs thanks for taking a look at this. I realized the `reverse` prop isn't actually documented for FormField and it is not included in the PropTypes.js so it makes sense it is not in the index.d.ts. 

As far as the error not showing up I think this is not because the `reverse` prop is valid but instead I think type checking is no longer happening. I tested locally with a type that should have failed and no error was showing up. I think this regression happened recently, I'm looking through some recent commits to see what may have happened
